### PR TITLE
Add cloud storage listBuckets sample.

### PR DESCRIPTION
**Pull request by jerjou**

### CODE REVIEW

File: storage/list_buckets.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
```
Comment (jeffmendoza) [inline storage/list_buckets.go]: Don't we need the full header?

File: storage/list_buckets_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"os"
+	"testing"
+)
+
+func TestXxx(t *testing.T) {
```
Comment (jeffmendoza) [inline storage/list_buckets_test.go]: Should be TestListBuckets

**Comment by jeffmendoza** [comment]
Assuming you want ListBuckets to be reusable, you might want to change it to pass on the errors to the caller: make the return value the slice and error. Someone copying this into their code might not want their program to exit on failure.

Ex:

``` go
func fooNum() (int, error) {
  num, err := someFunc()
  if err != nil {
    return 0, err
  }
  return num, nil
}
```

File: storage/list_buckets.go (hunk 1 above)
Comment (jerjou) [inline storage/list_buckets.go]: This is what I found in some other sources. *shrug. Full header can't hurt.

File: storage/list_buckets.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+func main() {
+	if buckets, err := ListBuckets(os.Getenv("TEST_PROJECT_ID")); err == nil {
+		// Print out the results
+		for _, bucket := range buckets {
+			fmt.Println(bucket.Name)
+		}
+	} else {
+		log.Fatal(err.Error())
+	}
+}
```
Comment (jeffmendoza) [inline storage/list_buckets.go]: change to err !=nil, put the log.Fatal first, and remove the else.

Oh, and you'll need to pull the ListBuckets call out of the if statement to access the buckets var outside the if context.

Comment (jerjou) [inline storage/list_buckets.go]: Doesn't 'buckets' go out of scope, then?

**Comment by jerjou** [comment]
FYI I renamed the files to 'listbuckets', as seems to be the convention, and upper-cased the START comment (ie listBuckets -> ListBuckets)

**Comment by jeffmendoza** [comment]
LGTM

**Pull request merged**